### PR TITLE
Make certificate validity checks clearer

### LINKED ISSUE #509: RejectExpired also rejects future-dated certificates
When RejectExpired is turned on in the config, Trillian not only rejects certificates with NotAfter in the past, it also rejects certificates with NotBefore in the future. This is a bit surprising because it contradicts the name, even though the behavior is documented in the config PB.

Jacob Hoffman-Andrews [12:30 PM]
It looks like it might just be a happenstance of using the x509 library's `DisableTimeChecks`:

```
verifyOpts := x509.VerifyOptions{                                                                                                                           
    Roots:             validationOpts.trustedRoots.CertPool(),                                                                                                
    CurrentTime:       now,                                                                                                                                   
    Intermediates:     intermediatePool.CertPool(),                                                                                                           
    DisableTimeChecks: !validationOpts.rejectExpired,
```

Would the Trillian team be open to change that to unconditionally set `DisableTimeChecks: true`, and then have a specific check that `NotAfter` is in the future, if `rejectExpired` is true? Right now the behavior of `rejectExpired` doesn't really match the name (though it does match the documentation). Also, forward-dating certificates is not technically illegal, and seems like a separate thing from expired certificates.

### PR DESCRIPTION
**Pull request by daviddrysdale**
Note that this changes the semantics of the log config proto.

Would fix #509

### CODE REVIEW

File: trillian/ctfe/cert_checker.go [hunk 1]
```diff
@@ -82,13 +82,19 @@ func ValidateChain(rawChain [][]byte, validationOpts CertValidationOpts) ([]*x50
...
-		}
-		if !now.Before(cert.NotBefore) && !now.After(cert.NotAfter) {
-			return nil, errors.New("only expired certificates are accepted")
-		}
+	if now.IsZero() {
+		now = time.Now()
+	}
+	expired := now.After(cert.NotAfter)
+	if validationOpts.rejectExpired && expired {
+		return nil, errors.New("only non-expired certificates are accepted")
```
[the remark this answers is not in the document]
Comment (daviddrysdale) [reply trillian/ctfe/cert_checker.go]: Good point, done.

**Comment by daviddrysdale** [comment]
@jsha, does this look like what you were after?

**Comment by jsha** [comment]
That looks great, thanks! I think you could probably leave out the
reject_future flag. I doubt anyone needs it.

**Comment by daviddrysdale** [comment]
Dropped final commit.

**Pull request merged**